cmd/dump.go

```diff
@@ -61,79 +168,113 @@ func fatal(format string, args ...interface{}) {
 }
 
 func runDump(ctx *cli.Context) error {
+	var file *os.File
+	fileName := ctx.String("file")
+	if fileName == "-" {
+		file = os.Stdout
+		err := log.DelLogger("console")
+		if err != nil {
+			fatal("Deleting default logger failed. Can not write to stdout: %v", err)
+		}
+	}
 	setting.NewContext()
+	// make sure we are logging to the console no matter what the configuration tells us do to
+	if _, err := setting.Cfg.Section("log").NewKey("MODE", "console"); err != nil {
+		fatal("Setting logging mode to console failed: %v", err)
+	}
+	if _, err := setting.Cfg.Section("log.console").NewKey("STDERR", "true"); err != nil {
+		fatal("Setting console logger to stderr failed: %v", err)
+	}
 	setting.NewServices() // cannot access session settings otherwise
 
 	err := models.SetEngine()
 	if err != nil {
 		return err
 	}
 
-	tmpDir := ctx.String("tempdir")
-	if _, err := os.Stat(tmpDir); os.IsNotExist(err) {
-		fatal("Path does not exist: %s", tmpDir)
-	}
-	tmpWorkDir, err := ioutil.TempDir(tmpDir, "gitea-dump-")
-	if err != nil {
-		fatal("Failed to create tmp work directory: %v", err)
+	if file == nil {
+		file, err = os.Create(fileName)
```

Comment:
If we're going to set the mode to 0600 below we should just create with mode 0600